webapi/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -213,49 +215,88 @@ public async Task<string> ExtractAudienceAsync(SKContext context, CancellationTo
     }
 
     /// <summary>
-    /// Extract chat history.
+    /// Method that wraps GetAllowedChatHistoryAsync to get allotted history messages as one string.
+    /// GetAllowedChatHistoryAsync optionally updates a ChatHistory object with the allotted messages,
+    /// but the ChatHistory type is not supported when calling from a rendered prompt, so this wrapper bypasses the chatHistory parameter.
     /// </summary>
-    /// <param name="context">Contains the 'tokenLimit' controlling the length of the prompt.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     [SKFunction, Description("Extract chat history")]
-    public async Task<string> ExtractChatHistoryAsync(
+    public Task<string> ExtractChatHistory(
         [Description("Chat ID to extract history from")] string chatId,
         [Description("Maximum number of tokens")] int tokenLimit,
         CancellationToken cancellationToken = default)
+    {
+        return this.GetAllowedChatHistoryAsync(chatId, tokenLimit, cancellationToken: cancellationToken);
+    }
+
+    /// <summary>
+    /// Extract chat history within token limit as a formatted string and optionally update the ChatCompletionContextMessages object with the allotted messages
+    /// </summary>
+    /// <param name="chatId">Chat ID to extract history from.</param>
+    /// <param name="tokenLimit">Maximum number of tokens.</param>
+    /// <param name="chatHistory">Optional ChatCompletionContextMessages object tracking allotted messages.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>Chat history as a string.</returns>
+    private async Task<string> GetAllowedChatHistoryAsync(
+        string chatId,
+        int tokenLimit,
+        ChatCompletionContextMessages? chatHistory = null,
```

Comment:
Why is this optional? This function can have two outputs depending on if this is passed in.

Reply:
I'm not sure I know what you're talking about. 

This is optional because we don't need to update a ChatHistory object when extracting chat history as a string during other steps like audience and user intent extraction, and it's the same logic to allocate the messages. But we can't call this method directly because SK functions don't allow non-native types as parameters, so ChatHistory isn't a valid parameter type. The wrapper `ExtractChatHistory` above is required so that `GetAllowedChatHistoryAsync` can be called from a rendered prompt.

Reply:
Aren't we using the chat completion model in user intent extraction and other places? Why is ChatHistory not needed?

Reply:
@TaoChenOSU We are using the chat completion model for all the initial hops to the LLM, but I opted out of refactoring those into the new prompt template format because they're not chat centric i.e., they don't need to be optimized for chat since the prompt template and user input for these don't demand a conversational bot response. 

Hence, it makes more sense to keep these in the format of text completion than chat completion, so ChatHistory isn't needed.